ipfx/hbg_dataset.py

```diff
@@ -7,10 +7,10 @@
 
 
 class HBGDataSet(EphysDataSet):
-    def __init__(self, sweep_info=None, nwb_file=None, ontology=None, api_sweeps=True):
+    def __init__(self, sweep_info=None, nwb_file=None, ontology=None, api_sweeps=True, validate_stim=True):
         super(HBGDataSet, self).__init__(ontology)
         self.nwb_data = nwb_reader.create_nwb_reader(nwb_file)
-
+        self.validate_stim = validate_stim
         if sweep_info is not None:
             sweep_info = self.modify_api_sweep_info(
                 sweep_info) if api_sweeps else sweep_info
```

Comment:
I'm not understanding the changes here. Currently only `EphysDataSet` uses self.validate_stim but that still does not set it. So if you want to add support for it in the derived classes, you can add support for it in the EphysDataset ctor and hand it down.

In the current  version `validate_stim` is passed into AIBSDataSet and HBGDataSet but not used.

Reply:
I can see how it could be confusing. Python is too forgiving of a language to allow this kind of code. I have followed your suggestions and set `validate_stim` in `EphysDataSet` ctor. Thanks.